### PR TITLE
[GSoC] sympy.fps return a formal power series object for polynomial functions

### LINKED ISSUE #12310: sympy.fps doesn't always return a formal power series
See the following:

```pycon
>>> from sympy import Symbol, fps
>>> x = Symbol('x')
>>> p = fps(x ** 2)
>>> p
x**2
>>> type(p)
<class 'sympy.core.power.Pow'>
>>> p[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'Pow' object does not support indexing
```

I assume the logic is that the value in question is already a polynomial so doesn't need expansion, but it doesn't support the same interface, and [the documentation](http://docs.sympy.org/dev/modules/series/formal.html#sympy.series.formal.fps) doesn't suggest that this is what is supposed to happen.

### PR DESCRIPTION
**Pull request by arighnac**
#### References to other Issues or PRs
Fixes #12310 (The issue which led to this PR)

Two former PR's --
Closes #13854
Closes #12324

#### Brief description of what is fixed or changed

**Formerly**, polynomial functions, when referenced through `fps`, did not return a `Formal Power Series` object, instead the function itself got returned.

```
>>> from sympy import Symbol, fps
>>> x = Symbol('x')
>>> p = fps(x ** 2)
>>> p
x**2
>>> type(p)
<class 'sympy.core.power.Pow'>
>>> p[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'Pow' object does not support indexing
```
I implemented the `Coeff` class according to the lines of Aaron's comment [here](https://github.com/sympy/sympy/issues/12310#issuecomment-286585230). Further integration of class methods and functions might be required into the class. Coefficient sequence of the formal power series of polynomial functions contains a `Coeff` object now.

**Now**, A `Formal Power Series` object gets returned.

```
>>> from sympy import Symbol, fps
>>> x = Symbol('x')
>>> p = fps(x ** 2)
>>> p
FormalPowerSeries(x**2 + x + 1, x, 0, 1, (SeqFormula(Coeff(x**2 + x + 1, x, _k), (_k, 1, oo)), SeqFormula(x**_k, (_k, 0, oo)), 1))
>>> p.truncate()
1 + x + x**2 + O(x**6)
>>> p[0]
1
```

#### Release Notes

* series
    * sympy.fps returns a `Formal Power Series` object for `polynomial` functions.

### CODE REVIEW

File: sympy/series/formal.py [hunk 1]
```diff
@@ -901,6 +905,16 @@ def compute_fps(f, x, x0=0, dir=1, hyper=True, order=4, rational=True,
     return _compute_fps(f, x, x0, dir, hyper, order, rational, full)
 
 
+class Coeff(Function):
+    """
+    Coeff(p, x, n) represents the nth coefficient of the polynomial p in x
+    """
+    @classmethod
+    def eval(cls, p, x, n):
+        if p.is_polynomial(x) and n.is_integer:
```
Comment (czgdp1807) [inline sympy/series/formal.py]: `n.is_integer` may not necessarily be Boolean. Is it possible to use `fuzzy_and`?

Comment (czgdp1807) [reply sympy/series/formal.py]: What will happen if n is a float? not a polynomial? Can this class be accessed by non-polynomials?

Comment (ArighnaIITG) [reply sympy/series/formal.py]: This class can't be accessed by non-polynomials. And moreover, FPS is a generalization of a polynomial, so `n` being float is out of question.

**Comment by ArighnaIITG** [comment]
ping @leosartaj

File: sympy/series/formal.py [hunk 2]
```diff
@@ -779,7 +779,11 @@ def _compute_fps(f, x, x0, dir, hyper, order, rational, full):
                 result[2].subs(x, rep2 + rep2b))
 
     if f.is_polynomial(x):
-        return None
+        k= Dummy('k')
```
Comment (leosartaj) [inline sympy/series/formal.py]: ``k = Dummy('k')``

**Review by leosartaj — APPROVED** [review]
Other than my one minor comment this looks good to me. Let's get the CI passed and then we can merge this.

**Comment by czgdp1807** [comment]
I think you forgot to add test in `test_args`. It happens a lot.

**Comment by ArighnaIITG** [comment]
@leosartaj . Changes made. Ready for merging !

**Comment by leosartaj** [comment]
Looks good to me. Will merge in 24hrs if no objection.

**Pull request merged**